Check uncommitted build files on pre-push

ybiquitous opened the pull request:
<!-- Each pull request must be associated with an open issue unless it's a documentation fix. If a corresponding issue does not exist, please create one so we can discuss the change first. -->

<!-- Please answer the following. We close pull requests that don't. -->

> Which issue, if any, is this issue related to?

None.

> Is there anything in the PR that needs further explanation?

This aims to prevent us from forgetting commits of built files (`*.cjs`).

Additionally, this lower the log level of Rollup.

Example:

```sh-session
$ git push
...
You must commit the following files:
lib/utils/matchesStringOrRegExp.cjs
husky - pre-push script failed (code 1)
error: failed to push some refs to 'https://github.com/stylelint/stylelint.git'
```

If you want to bypass the pre-push hook, use `--no-verify`:

```sh-session
$ git push --no-verify
```

Mouvedia commented:
Can you add a script—e.g. npm run foo—that accepts a file path or a list of file paths?
i.e. so that you can _build_ specific files (`npm run build` builds everything every time)

ybiquitous commented:
Unluckily, it doesn't make sense. We can make `npm run build` accept any specified files like this:

```js
// rollup.config.js
let inputFiles = (process.env.INPUT_FILES ?? '').split(/\s+/).filter(Boolean);

if (inputFiles.length === 0) {
	inputFiles = fg.globSync([
		'lib/**/*.mjs',
		'!**/__tests__/**',
		'!lib/testUtils/**',

		// NOTE: We cannot support CommonJS for `cli.mjs` since the `meow` dependency is pure ESM.
		'!lib/cli.mjs',
	]);
}
```

Then, run:

```sh
INPUT_FILES='lib/standalone.mjs' npm run build
```

This unexpectedly outputs CJS files, e.g.

```diff
diff --git a/lib/formatters/stringFormatter.cjs b/lib/formatters/stringFormatter.cjs
index a2fc1e61d..e9bc8b1f2 100644
--- a/lib/formatters/stringFormatter.cjs
+++ b/lib/formatters/stringFormatter.cjs
@@ -2,7 +2,7 @@
 // please instead edit the ESM counterpart and rebuild with Rollup (npm run build).
 'use strict';
 
-const node_path = require('node:path');
+const path = require('node:path');
 const process = require('node:process');
 const picocolors = require('picocolors');
 const stringWidth = require('string-width');
@@ -99,7 +99,7 @@ function logFrom(fromValue, cwd) {
 		return underline(fromValue);
 	}
 
-	const filePath = node_path.relative(cwd, fromValue).split(node_path.sep).join('/');
+	const filePath = path.relative(cwd, fromValue).split(path.sep).join('/');
 
 	return terminalLink(filePath, `file://${fromValue}`);
 }
```

In summary, we must build all the files at once.

Mouvedia commented:
Then maybe enable [`cache`](https://rollupjs.org/configuration-options/#cache)?

ybiquitous commented:
> The `cache` property of a previous bundle. Use it to speed up subsequent builds in watch mode

It seems available only on watch mode. Our `npm run build` doesn't speed up unluckily.
